core/src/lib.rs

```diff
@@ -37,18 +37,12 @@
 
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 
-#[allow(clippy::derive_partial_eq_without_eq)]
-mod keys_proto {
-    include!(concat!(env!("OUT_DIR"), "/keys_proto.rs"));
-}
-
-mod envelope_proto {
-    include!(concat!(env!("OUT_DIR"), "/envelope_proto.rs"));
-}
-
-#[allow(clippy::derive_partial_eq_without_eq)]
-mod peer_record_proto {
-    include!(concat!(env!("OUT_DIR"), "/peer_record_proto.rs"));
+mod proto {
+    include!("generated/mod.rs");
```

Comment:
Why do we need an `include` here? Wouldn't it be simpler to just declare the `generated` module as a normal module on the root level and then just re-export?

Reply:
Perhaps the `--single-mod` option helps us removing some of this boilerplate?

Reply:
My concern is devs importing from root or proto inconsistently. I can add it to root though, no problem.

Reply:
Ah that is a valid concern! Happy to stick to `include` then! I am quite surprised this works though. This just imports the content of the file right? Why doesn't rustc expect those modules to be within a `proto` directory when the modules are declared within a `proto` module?

Reply:
I think it's because include! parses the file as an expression.